### DISCUSSION TITLE
Video streaming using EdgeX

### DISCUSSION THREAD

**Discussion by neerajasjawali** [edgexfoundry/community#62] [Q&A] [answered]
Hi All,

I am working on an usb video capturing application using EdgeX(jakarta) with security. I use 2 cameras for capturing video of 16s(25fps, 30mb file size) and publish the captured video to subscriber using EdgeX MessageBus of type MQTT. 
I do not know why but the services are getting killed due to Out Of Memory. 

Can I use EdgeX for video streaming?
If yes how and if no why?

Thanks in advance

**Comment by lenny-goodell** [comment, marked as answer]
No, because it is too much data. 
Please see the latest Levski release which has the new Device USB Camera service. This service will create a RTSP stream for the camera(s) which can be consumed by an app running AI/ML models. The inference results can then be pushed into EdgeX for other apps to act on.
https://github.com/edgexfoundry/device-usb-camera/tree/v2.3.0
Also see the Camera Management App example here:
https://github.com/edgexfoundry/edgex-examples/tree/main/application-services/custom/camera-management
This currently only supports Device ONVIF Camera, but support for Device USB Camera is in progress.

**Reply by neerajasjawali** [reply]
I was working on it since many days, and couldn't find any reason for wrapping the things. It was really helpful. Thank you @lenny-intel.

**Reply by lenny-goodell** [reply]
Here are a couple Intel reference implementation that use EdgeX and CV . Both push the inference data from the modules into EdgeX and have app service(s) do the analyses. 
https://intel-iot-devkit.github.io/rtsf-at-checkout-reference-design/
https://intel-iot-devkit.github.io/automated-checkout/index.html

**Reply by neerajasjawali** [reply]
Thank you @lenny-intel
